pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -189,8 +189,10 @@ func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext)
 			return err
 		}
 
+		found := false
 		for _, pdbItem := range pdbs {
```

Comment:
how many `pdb` do we expect to get?

Reply:
mhm, but the name must be unique therefore we should only get/expect a single PDB, am I right?

Reply:
The reason to call `List()` and not `Get` is to avoid unnecessary communication with the apiserver. To my understand `List` only reads the cache and `Get` makes a direct call to the apiserver.

Reply:
```
// PodDisruptionBudgetNamespaceLister helps list and get PodDisruptionBudgets.
// All objects returned here must be treated as read-only.
type PodDisruptionBudgetNamespaceLister interface {
	// List lists all PodDisruptionBudgets in the indexer for a given namespace.
	// Objects returned here must be treated as read-only.
	List(selector labels.Selector) (ret []*v1.PodDisruptionBudget, err error)
	// Get retrieves the PodDisruptionBudget from the indexer for a given namespace and name.
	// Objects returned here must be treated as read-only.
	Get(name string) (*v1.PodDisruptionBudget, error)

}
```

:)

Reply:
willing to refactor?

Reply:
Done